src/runtime/entitlements-manager.js

```diff
@@ -244,26 +252,50 @@ export class EntitlementsManager {
       default:
         return;
     }
-    this.postEntitlementsRequest_(new EntitlementJwt(), result, source);
+    const token = parseQueryString(this.win_.location.search)['gaa_n'];
+    let isUserRegistered = null;
+    if (
+      'additionalParameters' in event &&
+      event.additionalParameters !== null &&
+      event.additionalParameters.getIsUserRegistered() !== null
+    ) {
+      isUserRegistered = event.additionalParameters.getIsUserRegistered();
+    }
+    this.postEntitlementsRequest_(
+      new EntitlementJwt(),
+      result,
+      source,
+      token,
+      isUserRegistered
+    );
   }
 
   // Informs the Entitlements server about the entitlement used
   // to unlock the page.
   postEntitlementsRequest_(
     usedEntitlement,
     entitlementResult,
-    entitlementSource
+    entitlementSource,
+    optionalToken = '',
+    optionalIsUserRegistered = null
   ) {
     const message = new EntitlementsRequest();
     message.setUsedEntitlement(usedEntitlement);
     message.setClientEventTime(toTimestamp(Date.now()));
     message.setEntitlementResult(entitlementResult);
     message.setEntitlementSource(entitlementSource);
+    message.setToken(optionalToken);
+    if (optionalIsUserRegistered !== null) {
```

Comment:
```suggestion
    if (typeof optionalIsUserRegistered === 'boolean') {
```

This might be better, it'd check that `optionalIsUserRegistered` is either `true` or `false`